### PR TITLE
*: Update to next libp2p version 0.24.0

### PR DESCRIPTION
**Pull request by mxinden**
~~For now here to test whether https://github.com/libp2p/rust-libp2p/pull/1698 fixes https://github.com/paritytech/polkadot/pull/1532.~~

~~Up until libp2p `v0.24.0` is released this can serve as a place to collect necessary downstream changes.~~

Update libp2p to [v0.24.0](https://github.com/libp2p/rust-libp2p/blob/master/CHANGELOG.md).

//CC @romanb @twittner @tomaka

### CODE REVIEW

**Comment by mxinden** [comment]
Looking at the various dashboards (CPU, memory, message count, ...) I can not tell a significant difference between nodes running 0.23 and 0.24 other than the connection count reduction through https://github.com/libp2p/rust-libp2p/pull/1698.


`kusama-sentry-ae1-0` and `kusama-sentry-ae1-1` have been previously running with libp2p https://github.com/libp2p/rust-libp2p/commit/e32ff74809746482b677e29e85407753cfb4f148 and are running libp2p 0.24.0 since 2020-08-19 ~11 a.m. CEST.

`kusama-sentry-ae1-0` runs with `--enable-authority-discovery` and thus makes excessive use of the DHT. The two nodes before yesterday 11 a.m. would never be on par in regards to connection count, but instead differ by ~100 connections, ignoring the periodic spikes triggered by authority discovery every 10 minutes. This is gone since libp2p 0.24.0.

image

Wild guess: Might this be related to https://github.com/libp2p/rust-libp2p/pull/1710/ which makes libp2p properly pass listen upgrade errors?

File: Cargo.lock [hunk 1]
```diff
@@ -2892,28 +2892,28 @@ dependencies = [
  "bytes 0.5.6",
  "futures 0.3.5",
  "lazy_static",
- "libp2p-core",
+ "libp2p-core 0.20.1",
```
Comment (tomaka) [inline Cargo.lock:2895]: You've probably forgotten to update a `Cargo.toml` somewhere. It seems that we're now using two different versions of libp2p at the same time.

Comment (mxinden) [reply Cargo.lock]: Good catch. I did a bad merge in 22cb019d7c3ee1fd4b23bc87b9959edb50714b86 not updating the libp2p version introduced in https://github.com/paritytech/substrate/pull/4954.

**Review by tomaka — APPROVED** [review]

**Review by twittner — APPROVED** [review]

**Comment by tomaka** [comment]
Please `bot merge` at your leisure.

**Comment by mxinden** [comment]
bot merge

**Pull request merged**